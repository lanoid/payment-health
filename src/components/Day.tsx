import * as React from 'react';
import data from '../data/evezy-transactions';
import DayProcessor from '../helpers/DayProcessor';

import { day } from '../types/day';

interface Props {
    day: day;
    dictionary: {
        successLabel: string;
        successesLabel: string;
        failureLabel: string;
        failuresLabel: string;
        noDataLabel: string;
        successClass: string;
        failureClass: string;
        noDataClass: string;
        months: string[];
    };
}

export default class Day extends React.Component<Props> {
    static defaultProps: Partial<Props> = {
        dictionary: {
            successLabel: 'success',
            successesLabel: 'successes',
            failureLabel: 'failure',
            failuresLabel: 'failures',
            noDataLabel: 'no data',
            successClass: 'success',
            failureClass: 'failure',
            noDataClass: 'no-data',
            months: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
        }
    }
    
    render () {
        const { day, dictionary } = this.props;
        const currentDay = DayProcessor(day, data, dictionary);
        return (
            <li className={`day ${currentDay.class}-${Math.abs(currentDay.score)} ${currentDay.monthLabel ? dictionary.months[currentDay.date.getMonth()].toLowerCase() : ''}`} title={`${currentDay.simpleDate} ${currentDay.type} ${currentDay.value} - ${currentDay.successes} ${dictionary.successesLabel}/${currentDay.failures} ${dictionary.failuresLabel}`}></li>
        )
    }
}