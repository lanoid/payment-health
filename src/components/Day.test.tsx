import * as React from 'react';
import { shallow } from 'enzyme';
import Day from './Day';

describe('Day', () => {
    const dayObj = {
        date: new Date(),
        monthLabel: true
    };
    const day = shallow(<Day key={1} day={dayObj} />);

    it('matches snapshot', () => {
        expect(day).toMatchSnapshot();
    });
});