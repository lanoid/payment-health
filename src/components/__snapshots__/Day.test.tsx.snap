// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day matches snapshot 1`] = `
<li
  className="day success-11 oct"
  title="29/10/2019 success - 20 successes/9 failures"
/>
`;
